src/Server/Workspace.php

```diff
@@ -200,4 +214,44 @@ public function xdependencies(): array
         }
         return $dependencyReferences;
     }
+
+    /**
+     * @param Options|null $settings
+     */
+    public function didChangeConfiguration(Options $settings = null)
+    {
+        if ($settings === null) {
+            return;
+        }
+
+        $changedOptions = $this->getChangedOptions($settings);
+        $this->options = $settings;
+
+        if (!empty(array_intersect($changedOptions, $this->options->getIndexerOptions()))) {
```

Comment:
This looks overly complicated. All it needs to check atm is the attribute `fileTypes`.